src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -22,32 +24,112 @@
+        List<Index> outOfBoundsIndexes = new ArrayList<>();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        for (int i = 0; i < targetIndexes.size(); i++) {
+            Index index = targetIndexes.get(i);
+            if (isValidIndex(index, lastShownList) == -1) {
+                outOfBoundsIndexes.add(index);
+            } else {
+                Task taskToDelete = lastShownList.get(index.getZeroBased());
+                model.deleteTask(taskToDelete);
+                deletedTasks.add(taskToDelete);
+                deletedTasksIndexes.add(index);
+            }
         }
-
-        Task taskToDelete = lastShownList.get(targetIndex.getZeroBased());
-        model.deleteTask(taskToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_TASK_SUCCESS, taskToDelete));
+        return result(deletedTasks, deletedTasksIndexes, outOfBoundsIndexes);
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof DeleteCommand // instanceof handles nulls
-                && targetIndex.equals(((DeleteCommand) other).targetIndex)); // state check
+                && targetIndexes.equals(((DeleteCommand) other).targetIndexes)); // state check
+    }
+
+    /**
+     * Converts the list of successfully deleted tasks into a string to be returned to the user
+     * @param deletedTasks
+     * @return
+     */
+    private String deletedTasksToString(List<Task> deletedTasks, List<Index> deletedTasksIndexes) {
+        String str = "Successfully Deleted Tasks: \n";
+        for (int i = deletedTasks.size() - 1; i >= 0; i--) {
+            str += deletedTasksIndexes.get(i).getOneBased() + ". " + deletedTasks.get(i) + "\n";
+        }
+        return str;
+    }
+
+    /**
+     * Converts a list of indexes to a string to be returned to the user.
+     * @param indexes
+     * @return
+     */
+    private String indexesToString(List<Index> indexes) {
+        String str = "" + indexes.get(0).getOneBased();
+        for (int i = 1; i < indexes.size(); i++) {
+            int index = indexes.get(i).getOneBased();
+            if (indexes.size() > 1 && i == indexes.size() - 1) {
+                str += " and " + index;
+            } else {
+                str += ", " + index;
+            }
+        }
+        return str;
+    }
+
+    /**
+     * Checks if index is a valid index to delete.
+     * If index > size of targetIndexes, return -1.
+     * If index is a valid index, return 0.
+     * @return an int representing validity of the index
+     */
+    private int isValidIndex(Index index, List<Task> taskList) {
+        if (index.getZeroBased() > taskList.size() - 1 || index.getZeroBased() < 0) { //index out of bounds
+            return -1;
+        } else { //valid index
+            return 0;
+        }
+    }
```

Comment:
I think it might be clearer if `isValidIndex` returns a `boolean` instead of an `int` since the method name sounds like a boolean!